src/vs/workbench/contrib/chat/browser/actions/chatActions.ts

```diff
@@ -439,6 +439,69 @@ abstract class ModeOpenChatGlobalAction extends OpenChatGlobalAction {
 	}
 }
 
+class SwitchCustomChatAgentAction extends OpenChatGlobalAction {
+	static readonly ID = 'workbench.action.chat.switchCustomChatAgent';
+
+	constructor() {
+		super({
+			id: SwitchCustomChatAgentAction.ID,
+			title: localize2('switchCustomChatAgent', "Switch to Custom Chat Agent"),
+			keybinding: undefined
+		});
+	}
+
+	override async run(accessor: ServicesAccessor, opts?: string | IChatViewOpenOptions): Promise<IChatAgentResult & { type?: 'confirmation' } | undefined> {
+		opts = typeof opts === 'string' ? { query: opts } : opts;
+
+		const chatModeService = accessor.get(IChatModeService);
+		const quickInputService = accessor.get(IQuickInputService);
+		const commandService = accessor.get(ICommandService);
+
+		const modes = chatModeService.getModes().custom;
+		if (modes.length === 0) {
+			return undefined;
+		}
+
+		let selectedModeName: string | undefined;
+		const agentName = opts?.mode;
+		if (agentName) {
+			// If agent name is provided as argument, use it directly
+			const selectedMode = modes.find(m => m.name === agentName);
+			selectedModeName = selectedMode?.name;
+		} else {
+			// Show quick pick to select - use a setTimeout to defer execution after accessor becomes invalid
+			const selected = await quickInputService.pick(
```

Comment:
I don't think we need an action with a separate quickpick for this case, users can already open the mode picker with the keybinding and type to pick a mode